crates/uv-resolver/src/lock/tree.rs

```diff
@@ -123,16 +126,27 @@ impl<'env> TreeDisplay<'env> {
                 }
             }
 
-            for (group, dependencies) in &packages.dev_dependencies {
+            for (group, dependencies) in &package.dev_dependencies {
                 for dependency in dependencies {
+                    // Skip dependencies that don't apply to the current environment.
+                    if let Some(environment_markers) = markers {
+                        if !dependency
+                            .complexified_marker
+                            .evaluate(environment_markers, &[])
+                        {
+                            skipped.insert(dependency.package_id.clone());
```

Comment:
Should we just add these to `non_roots` instead of `skipped`? Does it have the same effect?